### PR TITLE
[CENT-331]: Unify checkVariables and checkState

### PR DESCRIPTION
**Pull request by mirathewhite**
-Created checkMINTp0 that is specific for testing MINT p0 contracts.
-Created ControllerState and MintControllerState objects
-Replaced cloneState with 3rd party clone() function.

### CODE REVIEW

File: test/minting/ControllerTestUtils.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var BigNumber = require('bignumber.js');
 var Q = require('q');
+var clone = require('clone');
```
[the remark this answers is not in the document]
Comment (mirathewhite) [reply test/minting/ControllerTestUtils.js]: Deleting.

File: test/minting/MintControllerUtils.js [hunk 2]
```diff
@@ -13,42 +14,49 @@ var getAccountState = AccountUtils.getAccountState;
...
+    await checkState(mintController, customState, mintControllerEmptyState, getActualMintControllerState, Accounts, true);
 }
 
 
 // Gets the actual state of the mintController contract.
 // Evaluates all mappings on the provided accounts.
 async function getActualMintControllerState(mintController, accounts) {
-    return {
-        'minterManager': await mintController.minterManager.call()
+    var minterManager = await mintController.minterManager.call();
```
[the remark this answers is not in the document]
Comment (mirathewhite) [reply test/minting/MintControllerUtils.js]: The function `checkMintControllerState` calls `checkControllerState` in ControllerTestUtils.js`.  This lets us take advantage of inheritance: `MintController is Controller`.  So the `controllers` mapping and `owner` variable states are checked there.

**Comment by mirathewhite** [comment]
We discussed creating a special function `checkMINTp0` to force all unit tests to check both the `token` and the `mintController`.  I don't want to spend time making `checkVariables` more generic since it will go away eventually.  We plan to use the same framework for testing `token` as we do for the new contracts.

File: test/minting/MintControllerUtils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (walkerq) [reply test/minting/MintControllerUtils.js]: In that case can we remove the `accounts param` here?

**Pull request merged**